### ISSUE TITLE
`katib-db-manager` upgrade 1.6 to 1.7.1 unable to connect to DB

### ISSUE THREAD

**Issue by NohaIhab** [canonical/katib-operators#85]
Caught this during the pre-release testing of 1.7.1
Reproduce with an existing deployment of ckf 1.6/stable:
- `juju trust katib-db-manager --scope=cluster`
- `juju refresh katib-db-manager --channel 0.15/edge`

Result:
charm stuck in maintenance status with message: `Workload failed health check`

Pebble logs:
```
[katib-db-manager] F0609 09:40:58.029849    1014 main.go:104] Failed to open db connection: DB open failed: Timeout waiting for DB conn successfully opened.
[katib-db-manager] I0609 09:40:58.560760    1027 db.go:32] Using MySQL
[katib-db-manager] E0609 09:41:03.565105    1027 connection.go:40] Ping to Katib db failed: Error 1045: Access denied for user 'mysql'@'10.1.231.251' (using password: YES)
```

**Comment by NohaIhab** [comment]
This issue is to due to the user in the PodSpec version of `katib-db-manager` being [hardcoded](https://github.com/canonical/katib-operators/blob/katib-db-manager-rev66/charms/katib-db-manager/src/charm.py#L93), not grabbed from the relation data

**Comment by i-chvets** [comment]
In 1.6 Katib-db-manager podspec charm uses hardcoded root user, not user from relation. MaridDB charm is related via mysql relation using mysql interface.
When refreshing from 1.6 to 1.7.1 new sidecar charm uses username from its relation with MaridDB - mysql, which looks like it is not root (still need to confirm).
So, at no point new MySQL charm comes into play.
On clean 1.7.1 installations we should use new MySQL charm and relate via mysql_client:database interface. This should be all fine.
However, if on clean 1.7.1 installation client desires to use legacy MariaDB then we need to ensure MariaDB is configured with root user (that if we do revert back to hardcoding root).

**Comment by i-chvets** [comment]
1.6 katib-db-manager uses root username and root_password to access DB.
So, solution (depending on what exactly works) will be to either use hardcoded root user and password from relationrelation_data["root_password"] OR user/password from relationrelation_data["user"] and relation_data["password"]

**Comment by i-chvets** [comment]
Cannot reproduce the issue:
```
$ juju deploy katib-db-manager --channel 0.14/stable
Model     Controller          Cloud/Region        Version  SLA          Timestamp
kubeflow  microk8s-localhost  microk8s/localhost  2.9.42   unsupported  13:35:48-04:00

App               Version                Status  Scale  Charm                    Channel        Rev  Address        Exposed  Message
katib-db          mariadb/server:10.3    active      1  charmed-osm-mariadb-k8s  latest/stable   35  10.152.183.49  no       ready
katib-db-manager  res:oci-image@16b33a5  active      1  katib-db-manager         0.14/stable     66  10.152.183.27  no       

Unit                 Workload  Agent  Address      Ports     Message
katib-db-manager/0*  active    idle   10.1.59.120  6789/TCP  
katib-db/0*          active    idle   10.1.59.119  3306/TCP  ready

Relation provider  Requirer                Interface  Type     Message
katib-db:mysql     katib-db-manager:mysql  mysql      regular  

$ jhack show-relation katib-db:mysql     katib-db-manager:mysql
                                                 relation data v0.6                                                  
┏━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ relation (id: 0) ┃ katib-db                                      ┃ katib-db-manager                               ┃
┡━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ type             │ regular                                       │ =                                              │
│ interface        │ mysql                                         │ =                                              │
│ model            │ the current model                             │ =                                              │
│ relation ID      │ 0                                             │ =                                              │
│ role             │ provider                                      │ requirer                                       │
│ endpoint         │ mysql                                         │ mysql                                          │
│ leader unit      │ 0                                             │ 0                                              │
├──────────────────┼───────────────────────────────────────────────┼────────────────────────────────────────────────┤
│ application data │ ╭───────────────────────────────────────────╮ │ ╭────────────────────────────────────────────╮ │
│                  │ │ <empty>                                   │ │ │ <empty>                                    │ │
│                  │ ╰───────────────────────────────────────────╯ │ ╰────────────────────────────────────────────╯ │
│ unit data        │ ╭─ katib-db/0* ──────────────────╮            │ ╭─ katib-db-manager/0* ─╮                      │
│                  │ │                                │            │ │ <empty>               │                      │
│                  │ │  database       database       │            │ ╰───────────────────────╯                      │
│                  │ │  host           10.152.183.49  │            │                                                │
│                  │ │  password       password       │            │                                                │
│                  │ │  port           3306           │            │                                                │
│                  │ │  root_password  root           │            │                                                │
│                  │ │  user           mysql          │            │                                                │
│                  │ ╰────────────────────────────────╯            │                                                │
└──────────────────┴───────────────────────────────────────────────┴────────────────────────────────────────────────┘
```
Upgrade:
```
juju deploy katib-db-manager --channel 0.14/stable
$ juju status
Model     Controller          Cloud/Region        Version  SLA          Timestamp
kubeflow  microk8s-localhost  microk8s/localhost  2.9.42   unsupported  13:50:29-04:00

App               Version                Status  Scale  Charm                    Channel        Rev  Address         Exposed  Message
katib-db          mariadb/server:10.3    active      1  charmed-osm-mariadb-k8s  latest/stable   35  10.152.183.49   no       ready
katib-db-manager  res:oci-image@16b33a5  active      1  katib-db-manager         0.15/edge      239  10.152.183.241  no       

Unit                 Workload  Agent  Address      Ports     Message
katib-db-manager/0*  active    idle   10.1.59.121            
katib-db/0*          active    idle   10.1.59.119  3306/TCP  ready
$ jhack show-relation katib-db:mysql     katib-db-manager:mysql
                                                 relation data v0.6                                                  
┏━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ relation (id: 0) ┃ katib-db                                      ┃ katib-db-manager                               ┃
┡━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ type             │ regular                                       │ =                                              │
│ interface        │ mysql                                         │ =                                              │
│ model            │ the current model                             │ =                                              │
│ relation ID      │ 0                                             │ =                                              │
│ role             │ provider                                      │ requirer                                       │
│ endpoint         │ mysql                                         │ mysql                                          │
│ leader unit      │ 0                                             │ 0                                              │
├──────────────────┼───────────────────────────────────────────────┼────────────────────────────────────────────────┤
│ application data │ ╭───────────────────────────────────────────╮ │ ╭────────────────────────────────────────────╮ │
│                  │ │ <empty>                                   │ │ │ <empty>                                    │ │
│                  │ ╰───────────────────────────────────────────╯ │ ╰────────────────────────────────────────────╯ │
│ unit data        │ ╭─ katib-db/0* ──────────────────╮            │ ╭─ katib-db-manager/0* ─╮                      │
│                  │ │                                │            │ │ <empty>               │                      │
│                  │ │  database       database       │            │ ╰───────────────────────╯                      │
│                  │ │  host           10.152.183.49  │            │                                                │
│                  │ │  password       password       │            │                                                │
│                  │ │  port           3306           │            │                                                │
│                  │ │  root_password  root           │            │                                                │
│                  │ │  user           mysql          │            │                                                │
│                  │ ╰────────────────────────────────╯            │                                                │
└──────────────────┴───────────────────────────────────────────────┴────────────────────────────────────────────────┘

**Comment by i-chvets** [comment]
Was database upgraded first?

**Comment by NohaIhab** [comment]
@i-chvets database was refreshed to latest/stable as instructed in the upgrade guide, but got `charm is up to date` since it was just deployed before

**Comment by NohaIhab** [comment]
fixed by #88
